### PR TITLE
SobolIndicesAlgorithm doc clarified

### PR DESCRIPTION
**Pull request by josephmure**
A [discussion on Gitter](https://gitter.im/openturns/community?at=5f23bd57a2be904990815418) with @christelle-git revealed that the API doc of Sobol' indices-related classes could use some improvement. This PR rewords the API doc of the following classes (see [post below](https://github.com/openturns/openturns/pull/1562#issuecomment-671188945) for details).

- [SobolIndicesExperiment](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.SobolIndicesExperiment.html)
- [SobolIndicesAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.SobolIndicesAlgorithm.html)
- [SaltelliSensitivityAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.SaltelliSensitivityAlgorithm.html)
- [MartinezSensitivityAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.MartinezSensitivityAlgorithm.html)
- [MauntzKucherenkoSensitivityAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.MauntzKucherenkoSensitivityAlgorithm.html)
- [JansenSensitivityAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.JansenSensitivityAlgorithm.html)

Also, it increases the docstring example sample sizes in order to obtain valid results (negative Sobol' indices estimates were obtained before) and makes docstring examples across (Saltelli|Martinez|MauntzKucherenko|Jansen)SensitivityAlgorithm homogeneous.

It makes one change in the library: ``SobolSequence::MaxNumberOfDimension`` becomes ``SobolSequence::MaxNumberOfDimensions``.

### CODE REVIEW

**Comment by josephmure** [comment]
This PR is ready for review as is, but there are things I don't understand in the doc.

As far as I can tell, the notations at the end of the notes about [SaltelliSensitivityAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.SaltelliSensitivityAlgorithm.html#openturns.SaltelliSensitivityAlgorithm) are not defined anywhere. The same holds for Martinez, MauntzKucherenko and Jansen. I don't know what these quantities are @regislebrun @jschueller @mbaudin47 @sofianehaddad.

Saltelli_notes

**Comment by jschueller** [comment]
I dont remember if they initially came from a paper, but you can find these in
http://doc.openturns.org/papers/RT-PMFRE-01116-001C_-_Rapport_loi_estimateur_sobol.pdf

**Comment by josephmure** [comment]
Thanks @jschueller I'm going to read your document. Should I rebase to master in order to make Travis pass?

**Comment by josephmure** [comment]
Here is what the API Sobol indices-related doc looks like with this PR. I have used information provided by @mbaudin47 on [Gitter](https://gitter.im/openturns/community?at=5f2d1fbbfe39ca5d6587eecc). Feel free to correct what I may have misunderstood.

Notes section of [SobolIndicesExperiment](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.SobolIndicesExperiment.html):

SobolIndicesExperiment

Notes section of [SobolIndicesAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.SobolIndicesAlgorithm.html):

SobolIndicesAlgorithm1
SobolIndicesAlgorithm2
SobolIndicesAlgorithm3

[SaltelliSensitivityAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.SaltelliSensitivityAlgorithm.html):

SaltelliSensitivityAlgorithm

[MartinezSensitivityAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.MartinezSensitivityAlgorithm.html)

MartinezSensitivityAlgorithm

[MauntzKucherenkoSensitivityAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.MauntzKucherenkoSensitivityAlgorithm.html)

MauntzKucherenkoSensitivityAlgorithm

[JansenSensitivityAlgorithm](https://openturns.github.io/openturns/master/user_manual/_generated/openturns.JansenSensitivityAlgorithm.html)

JansenSensitivityAlgorithm

I wonder why 99% of Travis and Appveyor Python tests fail. It looks the same error is repeated over and over in all tests:

```
 11/553 Test  #489: pyinstallcheck_IndicesCollection_std ...................................***Failed    1.00 sec

Traceback (most recent call last):

  File "/Users/travis/build/openturns/openturns/python/test/t_IndicesCollection_std.py", line 4, in <module>

    import openturns as ot

  File "/Users/travis/.local/lib/python2.7/site-packages/openturns/__init__.py", line 84, in <module>

    from .simulation import *

  File "/Users/travis/.local/lib/python2.7/site-packages/openturns/simulation.py", line 678

SyntaxError: Non-ASCII character '\xc2' in file /Users/travis/.local/lib/python2.7/site-packages/openturns/simulation.py on line 679, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
```
Is it a Python 2 issue? I thought we had dropped Python 2 support in version 1.15.

**Comment by jschueller** [comment]
you got non ascii chars:
```
$ grep -P -n '[^\x00-\x7F]' ../python/src/*.i.in
../python/src/SobolIndicesAlgorithmImplementation_doc.i.in:129:Various estimators of :math:`V_i`, :math:`VT_i` and :math:`V_{-i}` are provided by the :class:`SobolIndicesAlgorithm` implementations:
../python/src/SobolIndicesExperiment_doc.i.in:54:    - ``'QMC'`` for a :class:`~openturns.LowDiscrepancyExperiment` (with *randomize* flag set to *False*) built from a :class:`~openturns.SobolSequence`.
```

opening it with a capable editor you'll you have utf8 nbsp's:
```
... :math:`VT_i`_and ...
... *randomize*_flag ...
```

**Comment by josephmure** [comment]
Thanks @jschueller! All tests pass now except one of the 2 Appveyor builds, but apparently it failed to download OpenBLAS, so the error is not related to the changes.

**Comment by josephmure** [comment]
Thanks @jschueller! All tests pass now except one of the 2 Appveyor builds, but apparently it failed to download OpenBLAS, so the error is not related to the changes.

File: lib/src/Base/Stat/SobolSequenceDirections.hxx [hunk 1]
```diff
@@ -22,10 +22,10 @@
  * which is covered by the LGPL license
  */
 
-const UnsignedInteger    SobolSequence::MaximumNumberOfDimension = 1111;
+const UnsignedInteger    SobolSequence::MaximumNumberOfDimensions = 1111;
```
Comment (jschueller) [inline lib/src/Base/Stat/SobolSequenceDirections.hxx:25]: I think we have maximumDimension or maximumIteration more often than maximumNumberOfXXX, maybe use MaximumDimension ?

Comment (josephmure) [reply lib/src/Base/Stat/SobolSequenceDirections.hxx]: Good idea. Done.

**Comment by josephmure** [comment]
Thank you for the approval @sofianehaddad. I think we should wait for @mbaudin47's review before merging. As I rewrote the doc based on what he wrote on Gitter, I think he should check everything is correct.

**Pull request merged**